src/material/button/button.ts

```diff
@@ -119,10 +119,8 @@ export class MatButton extends _MatButtonMixinBase
   }
 
   /** Focuses the button. */
-  focus(_origin?: FocusOrigin, options?: FocusOptions): void {
-    // Note that we aren't using `_origin`, but we need to keep it because some internal consumers
-    // use `MatButton` in a `FocusKeyManager` and we need it to match `FocusableOption`.
-    this._getHostElement().focus(options);
+  focus(origin: FocusOrigin = 'program', options?: FocusOptions): void {
```

Comment:
`FocusOrigin` has the definition `'touch' | 'mouse' | 'keyboard' | 'program' | null;` (Notice the `null`)

I think the correct(-er) fix would be to do:
```
focus(origin: FocusOrigin = null, options?: FocusOptions)
```
And let the `FocusMonitor` handle it however it wants to handle an unset `FocusOrigin` (which happens to be to treat it as 'program').

Reply:
I see your point. But I feel like explicitly having program as default is reasonable since this method intends to focus the button programmatically (unless a specific origin had been specified)